### PR TITLE
Use button elements as links

### PR DESCRIPTION
**Pull request by MartinEden**
This is a small change to allow users of the library to use Buttons as simple-react-router links.

### CODE REVIEW

File: src/Link.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Link extends Component {
   }
 
   onClick(event){
-    const href = this.refs.link.href
+    const href = this.refs.link.getAttribute("href");
```
Comment (jaredatron) [inline src/Link.js]: this all looks good by why this change?

**Review by jaredatron — CHANGES_REQUESTED** [review]

**Comment by jaredatron** [comment]
otherwise I love this change :)

**Comment by MartinEden** [comment]
Well asked - it turns out I didn't understand that point as well as I thought I did.

`href` is a property that only exists on `HTMLAnchorElement`, at least in the browsers I am testing. So I thought I could just use the more generally available `getAttribute` method to get the href property from the button.

However, it turns that `href` does a little more than just return the value of the href attribute - it also converts relative URLs to absolute ones. So I've now added another change that adds this behaviour back in now that I am using `getAttribute`.

However, I could understand it if you no longer liked this change, as it is a little more risky.

**Comment by MartinEden** [comment]
I'd love to hear your thoughts on this change, if you had time. Thanks! :)

File: src/Link.js [hunk 2]
```diff
@@ -27,8 +27,16 @@ class Link extends Component {
     this.onClick = this.onClick.bind(this);
   }
 
+  isAbsoluteURL(url) {
+    return url.startsWith("http://")
```
Comment (jaredatron) [inline src/Link.js:31]: I think a better way to check if a link is same domain or cross domain is:

```js
const link = this.refs.link;
location.hostname !== link.hostname;
```

because even if its an absolute url to the same domain we want to capture the click and use single-page-routing instead of a page reload.

Comment (jaredatron) [reply src/Link.js:31]: Also I think the `.href` property on a link DOM node will explode out relative paths to the full URL. If thats true in all browsers / cases than this will never be false.

Comment (jaredatron) [reply src/Link.js:31]: i think this proves that 

```js
const link = document.createElement('a'); link.href = "/foo/bar.jpg"; link.href
```

**Review by jaredatron — CHANGES_REQUESTED** [review]

**Comment by jaredatron** [comment]
otherwise I love this change and would like to merge it in :D

File: src/Link.js (hunk 2 above)
Comment (MartinEden) [reply src/Link.js:31]: Yes, but the problem is that I can't use the href property on the Button element, as it doesn't have it in most browsers. So instead I have to directly access the "href" attribute, which doesn't explode out relative paths.

So I am instead reintroducing that behaviour. I also don't have access to the `link.hostname` for comparison as I don't have access to the link object in the button, just a string.

Does that clarify things?

Comment (jaredatron) [reply src/Link.js:31]: yes that definitely clarifies things.

**Comment by jaredatron** [comment]
merges this manually. but made some subsequent changes. see master
